great_expectations/experimental/datasources/interfaces.py

```diff
@@ -335,22 +338,22 @@ class Datasource(
     }
     # Setting this in a Datasource subclass will override the execution engine type.
     # The primary use case is to inject an execution engine for testing.
-    execution_engine_override: ClassVar[Optional[Type[ExecutionEngine]]] = None
+    execution_engine_override: ClassVar[Optional[Type[_ExecutionEngineT]]] = None  # type: ignore[misc]
```

Comment:
What are the details behind this new type ignore? That is, I am curious what the error message is if you don't have this.

Reply:
Message: `ClassVar cannot contain type variables`. I added a comment in this PR: https://github.com/great-expectations/great_expectations/pull/7010